projects/packages/sync/src/replicastore/class-table-checksum.php

```diff
@@ -533,9 +533,22 @@ public function get_range_edges( $range_from = null, $range_to = null, $limit =
 
 		$this->validate_fields( array( $this->range_field ) );
 
+		// Performance :: When getting the postmeta range we do not want to filter by the whitelist.
+		// The reason for this is that it leads to a non-performant query that can timeout.
+		// Instead lets get the range based on posts regardless of meta.
+		$filter_valuess = $this->filter_values;
```

Comment:
`$filter_valuess` looks like a typo. Name it something like `$saved_filter_values` instead?

For that matter, code like this tends to make me wary versus something like
```php
$saved_filter_values = $this->filter_values;
$this->filter_values = null;
try {
    // `trim()` to make sure we don't add the statement if it's empty.
    $filters = trim( $this->build_filter_statement( $range_from, $range_to ) );
} finally {
    $this->filter_values = $saved_filter_values;
}
```
that won't forget to restore `$this->filter_values` in case something inside there throws. But maybe we don't need to care, if something throwing means the whole object should be considered invalid anyway.

Reply:
That's definitely a good approach that we want to adopt for the future! 

In this case, if it throws, we don't need to care as it will propagate further up and toss out the result anyway.

Reply:
> I wonder whether you might run into the same problem with commentmeta

As a note we have seen no indication that commentmeta has the same issue but that looks to be due to small number of keys in use. We will definitely conttinue to monitor but at this time it doesn't need any modifications.